include/deal.II/lac/vector_operations_internal.h

```diff
@@ -1248,11 +1252,13 @@ namespace internal
      */
     template <typename Operation, typename ResultType>
     void parallel_reduce (const Operation   &op,
-                          const size_type    vec_size,
+                          const size_type    end,
                           ResultType        &result,
-                          std_cxx11::shared_ptr<parallel::internal::TBBPartitioner> &partitioner)
+                          std_cxx11::shared_ptr<parallel::internal::TBBPartitioner> &partitioner,
+                          const size_type start = 0)
```

Comment:
Since this is an internal interface and user cannot cannot rely on this to remain constant, I would simply add `start` as the second argument. (If you agree, please apply the same change to the `parallel_for` part from #4015.) We do not need to stay backwards compatible here, in particular since this function is new for 8.5.